Bugfix and two new settings to toggle

dapal-003 opened the pull request:
- Fixed case insensitive not working with capitalized cute words, can't modify an object while iterating over it.
- Implemented setting to toggle cutecord for lurking guilds, the setting was never used from what I saw
- Added setting to get notified by cute words even if it originated from a muted guild
- Added setting to get notified even if you are focused on the channel the message originated from
- Clarified what "Lurking guilds" mean as before I've seen it refereed to as guilds you have muted.

dapal-003 committed:
fixed case insensetivity not working
clearified "lurked guilds" meaning
add option to overwrite muted guild supression
add option to notify even if chat is in focus
removed check that to my knowledge would never get satisfied
bumped version

dapal-003 committed:
Update README.md

katlyn commented on Settings.jsx:
```diff
@@ -227,7 +229,7 @@ module.exports = class Settings extends React.Component {
         </SwitchItem>
         <SwitchItem
-          note="Do you want to get notifications in guilds you're lurking?"
+          note="Do you want to get notifications in guilds you are lurking in? (Lurked servers are ones where you can't chat)"
```
Lurked is what this used to be called in the UI, it looks like the current terminology Discord uses is "preview mode". This should be updated to that as it's more accurate than "ones where you can't chat".

dapal-003 replied:
Thanks for pointing this out, I didn't know about a lurking mode to begin with so it was really confusing. Renamed to previewing

katlyn commented on index.js:
```diff
@@ -209,14 +209,16 @@ module.exports = class Cutecord extends Plugin {
     }
 
-    if (notificationSettings.allowNoMessages(channel)) {
-      // No.
+    const overwriteMuteSupression = this.settings.get('overwriteMuteSupression')
+    if (notificationSettings.allowNoMessages(channel) && !overwriteMuteSupression) {
+      // No. Unless we want them... I really do, don't judge me.
       return false
     }
```
This logic doesn't match your changelog note, instead it makes all notifications from muted channels go through (not just those that match the cute list).

dapal-003 replied:
Yes, this code does indeed let all muted notifications go through if the user decides to turn the feature on. However, later on in the method it would still need to pass a "cute" in order to return true. If it doesn't contain a cute it would check for DND and then finally drop it.

I've kept this check in place, but added an additional check after checking for cute, if there where no cute found and the message is muted, return that we don't want to get notified

katlyn commented on index.js:
```diff
@@ -54,14 +54,14 @@ module.exports = class Cutecord extends Plugin {
       (args) => {
-        const [{ message }] = args
-        if (message.originalMentioned === undefined) {
+        const [ { message } ] = args
+        if (!message.originalMentioned) {
```
You should not undo this strict type equality.

dapal-003 replied:
Didn't catch this auto ESLINT change, reverted back to strict type

katlyn commented:
Closes #14

katlyn commented on README.md:
```diff
@@ -33,6 +33,12 @@ If you have cutecord style notifications enabled, this is how we handle them.
 
 ---
 ## Changelog
+
+### v3.3.1
+Clearified what "Lurking guilds" Actually means, (No, it doesn't mean muted guilds!)
+Added option to still get notified by cuties even if they said something in a muted guild
+Added option to still get notified by cuties even if you can already see them (In the current chat)
+Fixed case insensitive not working with capatalized cute words
```
Match the style of the rest of the file please - no lines longer than 80 characters. In addition, displaying a newline in markdown requires two newlines. If you look at the preview on your branch, your whole changelog is one line. Preferably, use an unordered list as there are multiple changes.

dapal-003 replied:
Noted, I was editing the MD through a terminal at the time so didn't see it was all on one line. Re formatted it to use unordered lists and under 80char per line

dapal-003 committed:
Changed lurked termonoligy to preview
Changed logic for mute overwrite to require cute match
Undid ESLINT change to strict type equality
Updated changes to readme to fit the style

dapal-003 committed:
Update switch item tittle
Changed logic for muted notification
Messages that do not contain cute and are muted get blocked
